File: 11000s/11066.py
tc = int(input())
while(tc > 0):
    tc -= 1
    k = int(input())
    dp = [[0 for i in range(k+1)] for j in range(k+1)]
    arr = [0] + list(map(int, input().split()))
    _sum = [0] * (k+1)

    for i in range(1, k+1):
        _sum[i] = _sum[i-1] + arr[i]
    
    for i in range (1, k+1):
        for j in range (1, k-i+1):
            dp[j][j+i] = float('inf')
            for mid in range(j, i+j):
                mid_dp = dp[j][mid]+dp[mid+1][i+j]
                pre_sum = _sum[j+i] - _sum[j-1]
                dp[j][i+j] = min(dp[j][i+j],  mid_dp+pre_sum)

    print(dp[1][k])

File: 11000s/11066.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    int tc;
    cin >> tc;
    while(tc--)
    {
        int k;
        cin >> k;
        vector <vector <int>> dp(k+1);
        vector <int> arr(k+1);
        vector <int> sum(k+1);
        for(int i = 1; i <= k; i++)
        {
            dp[i].resize(k+1);
            cin >> arr[i];
        }
        for(int i = 1; i <= k; i++)
        {
            sum[i] = sum[i-1] + arr[i];
        }

        for(int i = 1; i <= k; i++)
        {
            for(int j = 1; j+i <= k; j++)
            {
                dp[j][j+i] = INT32_MAX;
                for(int mid = j; mid < i+j; mid++)
                {
                    int mid_dp = dp[j][mid]+dp[mid+1][i+j];
                    int pre_sum = sum[i+j] - sum[j-1];
                    dp[j][i+j] = min(dp[j][i+j], mid_dp+pre_sum);
                }
            }
        }

        cout << dp[1][k] << "\n";  
    }
}